lib/puppet/provider/iis_site/webadministration.rb

```diff
@@ -107,33 +163,36 @@ def self.prefetch(resources)
   def self.instances
     inst_cmd = ps_script_content('_getwebsites', @resource)
     result   = run(inst_cmd)
-    text     = result[:stdout]
-
-    site_json = JSON.parse(text)
-    site_json = [site_json] if site_json.is_a?(Hash)
-    site_json.collect do |site|
-      site_hash = {}
-
-      site_hash[:ensure]               = site['state'].downcase
-      site_hash[:name]                 = site['name']
-      site_hash[:physicalpath]         = site['physicalpath']
-      site_hash[:applicationpool]      = site['applicationpool']
-      site_hash[:serverautostart]      = to_bool(site['serverautostart'].downcase) unless site['serverautostart'].empty?
-      site_hash[:enabledprotocols]     = site['enabledprotocols']
-      site_hash[:logpath]              = site['logpath']
-      site_hash[:logperiod]            = site['logperiod']
-      site_hash[:logtruncatesize]      = site['logtruncatesize']
-      site_hash[:loglocaltimerollover] = to_bool(site['loglocaltimerollover'].downcase) unless site['loglocaltimerollover'].empty?
-      site_hash[:logformat]            = site['logformat']
-      site_hash[:logflags]             = site['logextfileflags']
-
-      new(site_hash)
+
+    unless result.nil? || result[:stdout].nil?
+      site_json = JSON.parse(result[:stdout])
+      site_json = [site_json] if site_json.is_a?(Hash)
+      unless site_json.nil?
+        site_json.collect do |site|
+          site_hash = {}
+
+          site_hash[:ensure]               = site['state'].downcase
+          site_hash[:name]                 = site['name']
+          site_hash[:physicalpath]         = site['physicalpath']
+          site_hash[:applicationpool]      = site['applicationpool']
+          site_hash[:serverautostart]      = to_bool(site['serverautostart'].downcase) unless site['serverautostart'].empty?
+          site_hash[:enabledprotocols]     = site['enabledprotocols']
+          site_hash[:logpath]              = site['logpath']
+          site_hash[:logperiod]            = site['logperiod']
+          site_hash[:logtruncatesize]      = site['logtruncatesize']
+          site_hash[:loglocaltimerollover] = to_bool(site['loglocaltimerollover'].downcase) unless site['loglocaltimerollover'].empty?
+          site_hash[:logformat]            = site['logformat']
+          site_hash[:logflags]             = site['logextfileflags']
+
+          new(site_hash)
+        end
+      end
     end
   end
 
   def self.to_bool(value)
-    return true   if value == true   || value =~ (/(true|t|yes|y|1)$/i)
-    return false  if value == false  || value =~ (/(false|f|no|n|0)$/i)
+    return true if value == true || value =~ (/(true|t|yes|y|1)$/i)
```

Comment:
Its generally bad form to change whitespace without change to code, harder to read diffs